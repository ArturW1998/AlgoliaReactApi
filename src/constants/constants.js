export const PATHS = {
  BASE_PATH: 'https://hn.algolia.com/api/v1',
  SEARCH_PATH: '/search',
  SEARCH_PARAM: 'query=',
  PAGE_HITS: 'hitsPerPage=',
  PAGE_PARAM: 'page=',
};

export const HITS = [
  {
    value: 10,
    label: 10,
  },
  {
    value: 20,
    label: 20,
  },
  {
    value: 40,
    label: 40,
  },
  {
    value: 50,
    label: 50,
  },
];

export const PAGES = {
  PREV: 'prev',
  NEXT: 'next',
};
