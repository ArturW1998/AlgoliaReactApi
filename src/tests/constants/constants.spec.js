import { PATHS, PAGES, HITS } from '../../constants/constants';

describe('Constants', () => {
  describe('PATHS constant', () => {
    it('should match the snapshot', () => {
      expect(PATHS).toMatchSnapshot();
    });
  });

  describe('PAGES constant', () => {
    it('should match the snapshot', () => {
      expect(PAGES).toMatchSnapshot();
    });
  });

  describe('HITS constant', () => {
    it('should match the snapshot', () => {
      expect(HITS).toMatchSnapshot();
    });
  });
});
