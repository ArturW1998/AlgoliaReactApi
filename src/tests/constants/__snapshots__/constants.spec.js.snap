// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Constants HITS constant should match the snapshot 1`] = `
Array [
  Object {
    "label": 10,
    "value": 10,
  },
  Object {
    "label": 20,
    "value": 20,
  },
  Object {
    "label": 40,
    "value": 40,
  },
  Object {
    "label": 50,
    "value": 50,
  },
]
`;

exports[`Constants PAGES constant should match the snapshot 1`] = `
Object {
  "NEXT": "next",
  "PREV": "prev",
}
`;

exports[`Constants PATHS constant should match the snapshot 1`] = `
Object {
  "BASE_PATH": "https://hn.algolia.com/api/v1",
  "PAGE_HITS": "hitsPerPage=",
  "PAGE_PARAM": "page=",
  "SEARCH_PARAM": "query=",
  "SEARCH_PATH": "/search",
}
`;
